### PR TITLE
pilot - added changelog validation

### LINKED ISSUE #11753: Check for changelog on externally distributed pilot builds
Distributing a TestFlight build to external testers will fail if a changelog is not specified.

Currently this fails after the lengthy build processing process like so:

```
[13:09:26]: Successfully finished processing the build 1.0 - 11
[13:09:26]: Distributing new build to testers: 1.0 - 11
[13:09:28]: Sending crash report...
[13:09:28]: The stack trace is sanitized so no personal information is sent.

[…snip…]

[13:09:28]: {"code"=>"ITC.apps.tf.build.test.info.missing.whats.new", "description"=>"Missing what's new info."}
```

We could check for this condition in the `upload` method:

https://github.com/fastlane/fastlane/blob/30bb1c4514a59c4ca2af4107b086e56408ece7c6/pilot/lib/pilot/build_manager.rb#L11-L53

e.g. if `options[:changelog]` is empty and `options[:distribute_external] == 'External'` then show a helpful `UI.user_error!` without making the user wait 20 minutes.

A side enhancement that could be done alongside this work is to load `changelog` from a metadata file, like [how _deliver_'s `UploadMetadata` class does it](https://github.com/fastlane/fastlane/blob/30bb1c4514a59c4ca2af4107b086e56408ece7c6/deliver/lib/deliver/upload_metadata.rb) (search for 'loader'). This would be helpful for CI tools that don't support parameterized builds (like CircleCI).

### PR DESCRIPTION
**Pull request by RishabhTayal**
fixes: #11753

### CODE REVIEW

**Comment by getaaron** [comment]
This looks great! I have a suggestion, let me know what you think. It might be easier to skim the code if we use the `unless` syntax like the line above.

**Comment by RishabhTayal** [comment]
@getaaron I am thinking instead of showing user error, should we show a prompt do users can provide changelog in run time? If we show error the user needs to rerun the lane just to provide changelog.

**Comment by getaaron** [comment]
@RishabhTayal That would be a good enhancement too. We should still provide an error message like what @krausefx mentioned in case fastlane is running in non interactive mode (eg through a CI server)

File: pilot/lib/pilot/build_manager.rb [hunk 1]
```diff
@@ -15,6 +15,10 @@ def upload(options)
 
       UI.user_error!("No ipa file given") unless config[:ipa]
 
+      if options[:changelog].nil? and options[:distribute_external] == true
+        UI.user_error!("No changelog provided")
```
[the remark this answers is not in the document]
Comment (RishabhTayal) [reply pilot/lib/pilot/build_manager.rb]: @KrauseFx refer to this: https://github.com/fastlane/fastlane/pull/11777#issuecomment-363113965. We might not need to have a error message.

**Comment by RishabhTayal** [comment]
@getaaron @KrauseFx Added user input and improved the error message. Ready for review. 👍

**Review by KrauseFx — APPROVED** [review]
Looks good to me, what do you think @getaaron

**Comment by RishabhTayal** [comment]
@KrauseFx What do you think about [this](https://github.com/fastlane/fastlane/issues/11753#issuecomment-362841962)? Is it doable?

**Review by getaaron — APPROVED** [review]

**Comment by getaaron** [comment]
@RishabhTayal Looks good to me, merging. 👍

Feel free to add a similar _deliver_ feature in a separate PR, I'm not sure if the requirements are identical though.

**Pull request merged**